exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -0,0 +1,241 @@
+	config *Config
+	logger *zap.Logger
+
+	startOnce sync.Once
+	client    *cloudwatchlogs.CloudWatchLogs // available after startOnce
+
+	seqTokenMu sync.Mutex
+	seqToken   string
+}
+
+func (e *exporter) Start(ctx context.Context, host component.Host) error {
+	var startErr error
+	e.startOnce.Do(func() {
+		awsConfig := &aws.Config{}
+		if e.config.Region != "" {
+			awsConfig.Region = aws.String(e.config.Region)
+		}
+		if e.config.Endpoint != "" {
+			awsConfig.Endpoint = aws.String(e.config.Endpoint)
+		}
+		if e.config.MaxRetries > 0 {
+			awsConfig.MaxRetries = aws.Int(e.config.MaxRetries)
+		}
+		sess, err := session.NewSession(awsConfig)
+		if err != nil {
+			startErr = err
+			return
+		}
+		e.client = cloudwatchlogs.New(sess)
+
+		e.logger.Debug("Retrieving Cloud Watch sequence token")
+		out, err := e.client.DescribeLogStreams(&cloudwatchlogs.DescribeLogStreamsInput{
+			LogGroupName:        aws.String(e.config.LogGroupName),
+			LogStreamNamePrefix: aws.String(e.config.LogStreamName),
+		})
+		if err != nil {
+			startErr = err
+			return
+		}
+		if len(out.LogStreams) == 0 {
+			startErr = errors.New("cannot find log group and stream")
+			return
+		}
+		stream := out.LogStreams[0]
+
+		e.seqTokenMu.Lock()
+		e.seqToken = *stream.UploadSequenceToken
+		e.seqTokenMu.Unlock()
+	})
+	return startErr
+}
+
+func (e *exporter) Shutdown(ctx context.Context) error {
+	// TODO(jbd): Signal shutdown to flush the logs.
+	return nil
+}
+
+func (e *exporter) ConsumeLogs(ctx context.Context, ld pdata.Logs) error {
+	logEvents, dropped := logsToCWLogs(e.logger, ld)
+	if len(logEvents) == 0 {
+		return nil
+	}
+	if dropped > 0 {
+		e.logger.Warn("CloudWatch Logs exporter dropped log records", zap.Any("count", dropped))
+	}
+
+	var seqToken string
+	e.seqTokenMu.Lock()
+	seqToken = e.seqToken
+	e.seqTokenMu.Unlock()
+
+	e.logger.Debug("Putting log events", zap.Int("num_of_events", len(logEvents)))
+	input := &cloudwatchlogs.PutLogEventsInput{
+		LogGroupName:  aws.String(e.config.LogGroupName),
+		LogStreamName: aws.String(e.config.LogStreamName),
+		LogEvents:     logEvents,
+		SequenceToken: aws.String(seqToken),
+	}
+	out, err := e.client.PutLogEvents(input)
+	if err != nil {
+		return err
+	}
+	if info := out.RejectedLogEventsInfo; info != nil {
+		return fmt.Errorf("log event rejected")
+	}
+	e.logger.Debug("Log events are successfully put")
+
+	// TODO(jbd): Investigate how the concurrency model of exporters
+	// impact the use of sequence tokens.
+	e.seqTokenMu.Lock()
+	e.seqToken = *out.NextSequenceToken
+	e.seqTokenMu.Unlock()
+	return nil
+}
+
+func logsToCWLogs(logger *zap.Logger, ld pdata.Logs) ([]*cloudwatchlogs.InputLogEvent, int) {
+	n := ld.ResourceLogs().Len()
+	if n == 0 {
+		return []*cloudwatchlogs.InputLogEvent{}, 0
+	}
+
+	var dropped int
+	out := make([]*cloudwatchlogs.InputLogEvent, 0) // TODO(jbd): set a better capacity
+
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
```

Comment:
This comment might have been confusing

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1944#discussion_r552299536

I meant we should convert `rl.Resource()` into `map` here and pass the map into `logToCWLog`. The resource is constant, no need to convert for each log in the same resource.